os/net/lwip/sys/arch/sys_arch.c

```diff
@@ -128,19 +129,25 @@ void sys_mbox_post(sys_mbox_t *mbox, void *msg)
 {
 	u8_t first_msg = 0;
 	u32_t tmp = 0;
-	sys_arch_sem_wait(&(mbox->mutex), 0);
-
+	u32_t status = OK;
+	status = sys_arch_sem_wait(&(mbox->mutex), 0);
+	if (status == SYS_ARCH_CANCELED) {
+		return;
+	}
 	LWIP_DEBUGF(SYS_DEBUG, ("mbox %p msg %p\n", (void *)mbox, (void *)msg));
 	/* Wait while the queue is full */
 	tmp = (mbox->rear + 1) % mbox->queue_size;
 	if (tmp == mbox->front) {
 		LWIP_DEBUGF(SYS_DEBUG, ("Queue Full, Wait until gets free\n"));
 	}
 	while (tmp == mbox->front) {
+		if (status == SYS_ARCH_CANCELED) {
```

Comment:
I think you had better put this at end of while, not beginnig of while.
When status is cancelled case, unncessary operation will be occured. (checking tmp == mbox->front)